tests/samplers_tests/test_samplers.py

```diff
@@ -48,6 +50,17 @@
 )
 
 
+def parametrize_suggest_method(name: str) -> MarkDecorator:
+    return pytest.mark.parametrize(
+        f"suggest_method_{name}",
+        [
+            lambda t: t.suggest_float(name, 0, 10),
+            lambda t: t.suggest_int(name, 0, 10),
+            lambda t: cast(float, t.suggest_categorical(name, [0, 1, 2])),
```

Comment:
[question] Why do you choose only those three suggest functions? Maybe, some samplers do not support other suggest functions?

Reply:
I just covered the modern `suggest` methods without arguments. This is mainly because the test time. If we cover the all six distributions, the time of the tests increase as follows:

- `test_single_parameter_objective`: 9.70s -> 19.83s
- `test_combination_of_different_distributions_objective` : 34.17s -> 139.99s
(CPU: AMD Ryzen 5 3600, Memory: 64GB)

The change of test code:
```diff
--- a/tests/samplers_tests/test_samplers.py
+++ b/tests/samplers_tests/test_samplers.py
@@ -57,6 +57,9 @@ def parametrize_suggest_method(name: str) -> MarkDecorator:
             lambda t: t.suggest_float(name, 0, 10),
             lambda t: t.suggest_int(name, 0, 10),
             lambda t: cast(float, t.suggest_categorical(name, [0, 1, 2])),
+            lambda t: t.suggest_float(name, 0, 10, step=0.5),
+            lambda t: t.suggest_float(name, 1e-7, 10, log=True),
+            lambda t: t.suggest_int(name, 1, 10, log=True),
         ],
     )
```

I think it's a bit long to take more than two minutes for a single test function. Do you have any ideas to reduce the test time while increasing the test coverage?

Reply:
`SkoptSampler` costs most of the test time. How about using random sampling in `skopt`? It is 5 times faster in my environment.

```python
-        lambda: optuna.integration.SkoptSampler(skopt_kwargs={"n_initial_points": 1}),
+        lambda: optuna.integration.SkoptSampler(skopt_kwargs={"base_estimator": "dummy", "n_initial_points": 1}),
```

Reply:
@not522 Thank you for your investigation. When I applied `"base_estimator" : "dummy"` to `SkoptSampler`, the test time greatly decreased as follows:
- `test_single_parameter_objective`: 19.83s -> 2.0s
- `test_combination_of_different_distributions_objective` : 139.99s -> 6.18s

I guess it is actually [implementation of random sampling](https://github.com/scikit-optimize/scikit-optimize/blob/273db8f3d223c4d471761e6d9ad1bf183b1ee6e1/skopt/optimizer/dummy.py), and we can use it in the unit test. I'm wondering if I should include the change since it may be out-of-scope of this PR. What do you think?

Reply:
I agree to keep the number of functions smaller in this PR.

Reply:
Since #2910 was merged into the master, I rebased the branch and introduced three additional suggest methods in https://github.com/optuna/optuna/pull/2869#discussion_r702512085. The test time is as follows:
- test_single_parameter_objective: 9.70s -> 19.83s -> 1.89s
- test_combination_of_different_distributions_objective : 34.17s -> 139.99s -> 6.03s

@keisuke-umezawa Please take another look.